libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteProgress.kt

```diff
@@ -0,0 +1,341 @@
+package com.mapbox.services.android.navigation.v5.routeprogress
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.geojson.Geometry
+import com.mapbox.geojson.Point
+import com.mapbox.navigator.BannerInstruction
+import com.mapbox.navigator.VoiceInstruction
+import com.mapbox.services.android.navigation.v5.utils.extensions.ifNonNull
+
+/**
+ * This class contains all progress information at any given time during a navigation session. This
+ * progress includes information for the current route, leg and step the user is traversing along.
+ * With every new valid location update, a new route progress will be generated using the latest
+ * information.
+ *
+ *
+ * The latest route progress object can be obtained through either the [ProgressChangeListener]
+ * or the [com.mapbox.services.android.navigation.v5.milestone.MilestoneEventListener] callbacks.
+ * Note that the route progress object's immutable.
+ *
+ *
+ * @since 0.1.0
+ */
+data class RouteProgress internal constructor(
```

Comment:
why `static builder` was removed? it's easy to use the same approach with koltin. Please take a look at `OfflineTiles` as a sample

Reply:
As I understand it, you want me to rewrite the builder. To what end?

Reply:
Let me show a diff.
New approach:
```
new RouteProgress().Builder()
```
old one:
```
RouteProgress.builder()
```
API was broken when we don't have to do it. To keep builder static without need to create an instance directly just move `builder()` fun to `companion object` and mark this function by `@JvmStatic` annotation. Just like in `OfflineTiles` class

Reply:
In any case, whatever we decide we should be consistent across the project.

Reply:
As discussed in our sync, the problem here is this call: toBuilder(). Look at the class MapboxNavigationNotificationTest(), method call checksInstructionTextIsUpdatedIfInstructionTextIsNotInitialized(). In this call buildDefaultTestRouteProgress() returns an object of type RouteProgress, not Builder. From that object the method toBuilder() is called. It must return the correct instance of the previously built RouteProgress object. But we don't have access to the Builder object at this point. We have access only to the RouteProgress object. When the Builder is called to create RouteProgress, it stores itself in the object that it built, thus toBuilder can return the correct instance.